src/arch/xtensa/drivers/idc.c

```diff
@@ -179,14 +179,18 @@ static int idc_pipeline_trigger(uint32_t cmd)
 	if (!pcm_dev)
 		return -ENODEV;
 
-	/* check whether we are executing from the right core */
-	if (arch_cpu_get_id() != pcm_dev->cd->pipeline->ipc_pipe.core)
-		return -EINVAL;
-
 	/* invalidate pipeline on start */
-	if (cmd == COMP_TRIGGER_START)
+	if (cmd == COMP_TRIGGER_START) {
+		/* invalidate cd before accessing it */
+		dcache_invalidate_region(pcm_dev->cd, sizeof(*pcm_dev->cd));
+
 		pipeline_cache(pcm_dev->cd->pipeline,
 			       pcm_dev->cd, CACHE_INVALIDATE);
+	}
+
+	/* check whether we are executing from the right core */
+	if (arch_cpu_get_id() != pcm_dev->cd->pipeline->ipc_pipe.core)
```

Comment:
Driver code should use the cpu_id() call like the next patch in this series does, although it may be better to create a new inline for this check since it's done a lot
```c
static inline bool pipeline_this_cpu(struct pipeline *p)
{
    if (p->ipc_pip.core == cpu_get_id(())
        return true;
    return false;
}
```

Reply:
Done.